In [39]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np


file = r"C:\Users\ambsc\Downloads\balanced_ai_human_prompts.csv"
df = pd.read_csv(file)

df['text_length'] = df['text'].astype(str).apply(len)        
df['word_count'] = df['text'].astype(str).str.split().str.len()

print("Dataset shape:", df.shape)
print("\nColumn types:\n", df.dtypes)
print("\nMissing values:\n", df.isna().sum())
print("\nCounts of AI vs Human:\n", df['generated'].value_counts())
print("\nSummary stats\n", df[['text_length','word_count']].describe())


Dataset shape: (2750, 4)

Column types:
 text           object
generated       int64
text_length     int64
word_count      int64
dtype: object

Missing values:
 text           0
generated      0
text_length    0
word_count     0
dtype: int64

Counts of AI vs Human:
 generated
1    1375
0    1375
Name: count, dtype: int64

Summary stats
        text_length   word_count
count  2750.000000  2750.000000
mean   1670.922909   290.771273
std    1654.118253   291.090474
min      78.000000    13.000000
25%      95.000000    14.000000
50%    1896.500000   304.500000
75%    2987.000000   525.000000
max    8436.000000  1360.000000


Data Information:

Source - Kaggle.com

Name - Human Vs AI Generated Essays

Uploader - Navjot Kaushal · Updated a month ago

Navjot Kaushal · Updated a month ago 


The dataset contains 4 variables being text, generated, text length, and word count. It has 2,750 observations

The text column is stored as strings while generated, text length, and word count are integers

There are no missing values

The dataset is  balanced with 1,375 AI-generated prompts and 1,375 human-written prompts

The text length ranges from 78 to 8,436 characters, with a mean of 1,670.9 characters

The word count ranges from 13 to 1,360 words, with a mean of 290.8 words

The median text length is 1,896.5 characters, but it includes some prompts that are extremely long which could skew the data



In [38]:
mean_ai = df[df['generated']==1]['text_length'].mean()
mean_human = df[df['generated']==0]['text_length'].mean()
diff = mean_ai - mean_human

print(f"Average text length (AI): {mean_ai:.1f} characters")
print(f"Average text length (Human): {mean_human:.1f} characters")
print(f"Difference (AI - Human): {diff:.1f} characters")


Average text length (AI): 169.3 characters
Average text length (Human): 3172.6 characters
Difference (AI - Human): -3003.3 characters


Average text length for AI generated prompts: 169.3 characters

Average text length for human prompts: 3,172.6 characters

Difference (AI - Human): -3,003.3 characters

From this we can conclude that human-written prompts are much longer on average than AI generated prompts

In [36]:
subset = df[['text','generated','text_length','word_count']]
subset.to_csv("balanced_ai_human_prompts_subset.csv", index=False)
print("Subset saved as 'balanced_ai_human_prompts_subset.csv'")


Subset saved as 'balanced_ai_human_prompts_subset.csv'
